src/services/user/hooks/registrationPins.js

```diff
@@ -1,62 +1,83 @@
 const hooks = require('feathers-hooks-common');
 const { authenticate } = require('@feathersjs/authentication');
-const logger = require('../../../logger');
+const { BadRequest, Forbidden } = require('@feathersjs/errors');
+const { Configuration } = require('@schul-cloud/commons');
 const globalHooks = require('../../../hooks');
 const pinModel = require('../../user/model').registrationPinModel;
 
 const removeOldPins = (hook) => pinModel.deleteMany({ email: hook.data.email })
-	.then((pins) => Promise.resolve(hook));
+	.then(() => Promise.resolve(hook));
 
 const generatePin = (hook) => {
 	const pin = Math.floor((Math.random() * 8999) + 1000);
 	hook.data.pin = pin.toString();
 	return Promise.resolve(hook);
 };
 
+const shortTitle = process.env.SC_SHORT_TITLE || 'Schul-Cloud*';
+const longTitle = process.env.SC_TITLE || 'HPI Schul-Cloud*';
 function createinfoText(hook) {
-	let text;
 	const role = hook.data.mailTextForRole;
 	const { pin } = hook.data;
-	const shortTitle = process.env.SC_SHORT_TITLE || 'Schul-Cloud*';
-	const longTitle = process.env.SC_TITLE || 'HPI Schul-Cloud*';
-	if (!role || !pin) {
-		logger.warning('Role or PIN missing to define mail text.');
-		return '';
+	if (!pin) {
+		throw new BadRequest('Fehler beim Erstellen der Pin.');
 	}
 	if (role === 'parent') {
-		text = `Vielen Dank, dass Sie Ihrem Kind durch Ihr Einverständnis die Nutzung der ${longTitle} ermöglichen.
-Bitte geben Sie folgenden Code ein, wenn Sie danach gefragt werden, um die Registrierung abzuschließen:
+		return `Vielen Dank, dass Sie Ihrem Kind durch Ihr Einverständnis die Nutzung der ${longTitle} ermöglichen.
+Bitte geben Sie den folgenden Bestätigungscode im Registrierungsprozess ein, um die Registrierung abzuschließen:
 
 PIN: ${pin}
 
 Mit Freundlichen Grüßen
 Ihr ${shortTitle} Team`;
-	} else if (role === 'student' || role === 'employee' || role === 'expert') {
-		text = `Vielen Dank, dass du die ${longTitle} nutzen möchtest.
-Bitte gib folgenden Code ein, wenn du danach gefragt wirst, um die Registrierung abzuschließen:
+	}
+	if (role === 'student' || role === 'employee' || role === 'expert') {
+		return `Vielen Dank, dass du die ${longTitle} nutzen möchtest.
+Bitte gib den folgenden Bestätigungscode im Registrierungsprozess ein, um deine Registrierung bei der ${longTitle} abzuschließen:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 129. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5203912)